gateway/src/http/mod.rs

```diff
@@ -295,12 +295,21 @@ impl<'a> HttpApiBuilder<'a> {
             .ok()
         });
 
+        let bind_address = self.gateway_config.http.bind_address;
         let router = nym_node_http_api::NymNodeRouter::new(config, None, wg_state);
 
-        let server = router
-            .build_server(&self.gateway_config.http.bind_address)?
-            .with_task_client(task_client);
-        tokio::spawn(async move { server.run().await });
+        tokio::spawn(async move {
+            let server = match router.build_server(&bind_address).await {
+                Ok(server) => server.with_task_client(task_client),
+                Err(err) => {
+                    error!("failed to create http server: {err}");
+                    // this will cause global shutdown
+                    drop(task_client);
```

Comment:
Does it not work without it? Was thinking it would be dropped at the end of the scope anyway?

Reply:
good point. I think I just wanted to be explicit to state that creation of the server will cause everything else to shutdown, but I think you're right. it's a bit redundant